### ISSUE TITLE
[question] How can indexed variables be used as indices?

### ISSUE THREAD

**Issue by alecjacobson** [iheartla/iheartla#86]
```
given
I ∈ ℤ^(2×2)
J ∈ ℤ^(2×2)

I_(J_1,1),(J_2,2)
```
doesn't compile:
```
Error on line 5 at column 3. Failed to parse integer: Expecting <digit>
I_(J_1,1),(J_2,2)
  ^
```

In matlibberish, I'd write this as `I(J(1,1),J(2,2))`. How can I achieve this in I❤️LA? 

Even introduce auxiliary variables didn't work (is this a bug):
```
given
I ∈ ℤ^(2×2)
J ∈ ℤ^(2×2)

a = J_1,1
b = J_2,2
I_a,b
```

**Comment by yig** [comment]
This seems partly related to shadowing the identity matrix. If I change the `I` to a `K`, the second version works:

```
given
K ∈ ℤ^(2×2)
J ∈ ℤ^(2×2)

a = J_1,1
b = J_2,2
K_a,b
```

[Try it](https://iheartla.github.io/iheartla/?code=given%0AK+%E2%88%88+%E2%84%A4%5E%282%C3%972%29%0AJ+%E2%88%88+%E2%84%A4%5E%282%C3%972%29%0A%0Aa+%3D+J_1%2C1%0Ab+%3D+J_2%2C2%0AK_a%2Cb)

The first version still doesn't.

**Comment by alecjacobson** [comment]
![image](https://user-images.githubusercontent.com/2241689/118990344-cb0bc300-b950-11eb-815b-fadfdd3a1e88.png)
on the web version linked in the "Try It"

**Comment by alecjacobson** [comment]
We can track the `I` shadowing via a new issue: https://github.com/iheartla/iheartla/issues/89

**Comment by yig** [comment]
The Try It link is fixed and it does indeed work when using K instead of I. (Apparently I❤️LA code must end with a blank line.)

**Comment by alecjacobson** [comment]
So is this fixed now and we can wait on #89 ?

**Comment by yig** [comment]
The first version even without shadowing the identity matrix doesn't work:

```
K ∈ ℤ^(2×2)
J ∈ ℤ^(2×2)

K_(J_1,1),(J_2,2)
```

I get:

```
Error on line 4 at column 3. Failed to parse integer: Expecting
K_(J_1,1),(J_2,2)
^
```

[Try it](https://iheartla.github.io/iheartla/?code=K+%E2%88%88+%E2%84%A4%5E%282%C3%972%29%0AJ+%E2%88%88+%E2%84%A4%5E%282%C3%972%29%0A%0AK_%28J_1%2C1%29%2C%28J_2%2C2%29%0A)
